File: tests/smoke/fastapi-sqlalchemy/app/services/order_service.py
from app.repositories.order_repository import OrderRepository


class OrderService:
    def __init__(self) -> None:
        self.repo = OrderRepository()

    def find_by_id(self, order_id: int):
        return self.repo.find_by_id(order_id)

    def create(self, dto: dict):
        return self.repo.insert(dto)

File: tests/smoke/nestjs-prisma/src/order/order.service.ts
import { Injectable } from '@nestjs/common';
import { OrderRepository } from './order.repository';

@Injectable()
export class OrderService {
  constructor(private readonly repo: OrderRepository) {}

  findById(id: number) {
    return this.repo.findById(id);
  }

  create(dto: Record<string, unknown>) {
    return this.repo.insert(dto);
  }
}
